### PR TITLE
Enhancement/8838 Use subtle notification component

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #8838 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/notifications/SubtleNotification.js
Comment (nfmohit) [inline assets/js/components/notifications/SubtleNotification.js:1]: It looks like the `Grid` wrapper in this component is resulting in the notification having an unexpected horizontal padding.

Ideally, a re-usable component should not use `Grid`, `Row`, and `Cell` as wrappers. I would suggest replacing these with a simple `div`, and refactoring the usage of this component in `assets/js/modules/reader-revenue-manager/components/dashboard/RRMSetupSuccessSubtleNotification.js` such that it is wrapped with the `Grid`, `Row`, and `Cell`.

What do you think?

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for your excellent work on this, @ankitrox. I have left a few comments, please let me know if you have any questions.

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-publication-onboarding-state-notice.scss
[the remark this answers is not in the document]
Comment (ankitrox) [reply assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-publication-onboarding-state-notice.scss:1]: On small screen, `SubtleNotification` hides the icon, but we need to display it for our component, so I kept it that single rule in this stylesheet. All other things are not required, so removed.

[the remark this answers is not in the document]
Comment (ankitrox) [reply assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-publication-onboarding-state-notice.scss:1]: Actually, I realised that some more rules are required, so added in the file.

File: assets/js/components/notifications/SubtleNotification.js
Comment (ankitrox) [reply assets/js/components/notifications/SubtleNotification.js:1]: I think that is a valid point. I've removed the use of Grid, Row, and Cell as wrappers from `SubtleNotification` component and added those wrappers in `assets/js/modules/reader-revenue-manager/components/dashboard/RRMSetupSuccessSubtleNotification.js`.

**Comment by ankitrox** [comment]
@nfmohit I've made the suggested changes. Please note that because we have made changes in `SubtleNotification` component, I had to make changes in couple of more places like in `RRMSetupSuccessSubtleNotification`, `SettingsEdit` and `SettingsView`. Also, added an additional CSS rule for `SettingsEdit` to separate select and notice component with the margin.

Assigning this to you for review. Please let me know in case of any query.

Thanks

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-publication-onboarding-state-notice.scss
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-publication-onboarding-state-notice.scss:1]: With the recent updates to the `SubtleNotification` component, I believe this stylesheet is no longer necessary. Could you reevaluate?

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for your work and patience on this, @ankitrox! Could you please update the branch with `develop`? Besides, I've also left a comment below. Thanks!

File: assets/js/components/notifications/SubtleNotification.js [hunk 1]
```diff
@@ -47,76 +46,70 @@ function SubtleNotification( {
...
+				'googlesitekit-subtle-notification',
+				{
+					'googlesitekit-subtle-notification--success':
+						variant === VARIANTS.SUCCESS,
+					'googlesitekit-subtle-notification--warning':
+						variant === VARIANTS.WARNING,
+				},
+				className
+			) }
+		>
```
Comment (nfmohit) [inline assets/js/components/notifications/SubtleNotification.js:70]: Thank you for making this change. It looks like this change has removed the top margin of the publication onboarding state notice in settings view. We will need to add that as part of the CSS styles.

File: assets/js/components/notifications/SubtleNotification.js [hunk 2]
```diff
@@ -47,76 +46,70 @@ function SubtleNotification( {
 	Icon,
 	ctaLink,
 	ctaLabel,
+	className,
 	onCTAClick,
 	isCTALinkExternal,
+	isDismissible = true,
```
Comment (nfmohit) [inline assets/js/components/notifications/SubtleNotification.js:52]: On further thought, I think we can achieve this without introducing a new prop and just checking if `dismissLabel` is defined, similar to how we check if `ctaLabel` is defined.

File: assets/js/components/notifications/SubtleNotification.js [hunk 3]
```diff
@@ -47,76 +46,70 @@ function SubtleNotification( {
...
 	dismissLabel = __( 'Ok, got it', 'google-site-kit' ),
```
Comment (nfmohit) [inline assets/js/components/notifications/SubtleNotification.js:53]: If we rely on if `dismissLabel` is defined, we need to remove the default value here. This should be fine as all current usages of this component (including stories) already specify this prop value.

```suggestion
	dismissLabel,
```

File: assets/js/components/notifications/SubtleNotification.js [hunk 4]
```diff
@@ -47,76 +46,70 @@ function SubtleNotification( {
...
-	onDismiss,
+	onDismiss = () => {},
```
Comment (nfmohit) [inline assets/js/components/notifications/SubtleNotification.js:54]: On further thought, I don't think this default value is required as well.

```suggestion
	onDismiss,
```

File: assets/js/modules/reader-revenue-manager/components/dashboard/RRMSetupSuccessSubtleNotification.js [hunk 5]
```diff
@@ -98,6 +99,8 @@ function RRMSetupSuccessSubtleNotification() {
 				publicationOnboardingState
 			);
 		}
+
+		dismissNotice();
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/dashboard/RRMSetupSuccessSubtleNotification.js:103]: It looks like this was mistakenly added back from the merge conflict. This was removed as part of a recent PR.

```suggestion
```

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-setup-module.scss [hunk 6]
```diff
@@ -27,6 +27,13 @@
 		.googlesitekit-publication-onboarding-state-notice {
 			margin-bottom: 40px;
 		}
+
+		&.googlesitekit-setup-module-edit {
+
+			.googlesitekit-publication-onboarding-state-notice {
+				margin-top: 16px;
+			}
+		}
```
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-setup-module.scss:36]: This is no longer needed anymore as the publication select now has horizontal spacing around it. Instead, we need some top spacing for this in the settings view screen.

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-setup-module.scss [hunk 7]
```diff
@@ -27,6 +27,13 @@
 		.googlesitekit-publication-onboarding-state-notice {
 			margin-bottom: 40px;
 		}
```
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-setup-module.scss:29]: This should only be applied to the module setup screen.

File: assets/js/modules/reader-revenue-manager/components/settings/SettingsEdit.js [hunk 8]
```diff
@@ -115,7 +115,7 @@ export default function SettingsEdit() {
 	}
 
 	return (
-		<div className="googlesitekit-setup-module googlesitekit-setup-module--reader-revenue-manager">
+		<div className="googlesitekit-setup-module googlesitekit-setup-module-edit googlesitekit-setup-module--reader-revenue-manager">
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/settings/SettingsEdit.js:118]: This additional class name should not be needed as we should be able to determine the screen based on already available selectors.

File: assets/js/modules/reader-revenue-manager/components/settings/SettingsView.js [hunk 9]
```diff
@@ -71,7 +71,7 @@ export default function SettingsView() {
 	} );
 
 	return (
-		<div className="googlesitekit-setup-module googlesitekit-setup-module--reader-revenue-manager">
+		<div className="googlesitekit-setup-module googlesitekit-setup-module-view googlesitekit-setup-module--reader-revenue-manager">
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/settings/SettingsView.js:74]: This additional class name should not be needed as we should be able to determine the screen based on already available selectors.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for your excellent work on this, @ankitrox! I've pointed out a few suggestions and issues in the PR.

For time constraints, I'm going to go ahead and make the updates myself. I hope that is okay with you.

Thank you!

**Comment by nfmohit** [comment]
Note: The failing VRT reference is unrelated to this PR.

**Pull request merged**